ext/curl/curl.stub.php

```diff
@@ -26,6 +26,27 @@ final class CurlShareHandle
 {
 }
 
+#if LIBCURL_VERSION_NUM >= 0x073E00 /* Available since 7.62.0 */
+/**
+ * @strict-properties
+ * @not-serializable
+ */
+final class CurlUrl implements Stringable
+{
+    public function __construct(?string $url = null) {}
+
+    public function get(int $part, int $flags = 0): string {}
```

Comment:
I think this would be our first chance to get rid of the legacy habit of using integers as options and expecting users to pass (global) constants. We already have enums which would offer a cleaner and easier-to-use API IMO. @iluuu1994 
```suggestion
    public function get(CurlUrlPart $part, int $flags = 0): string {}
```

I know doing so would introduce an inconsistency with the rest of ext/curl, but I think we should start the API cleanup somewhere :)

Reply:
Since `CurlUrl` is a final class, I think it would make sense to add the methods Ilija suggested (`getPath()` etc.). Otherwise people will always have to use composition in order to have a more meaningful, non-minimal API.

Reply:
Sorry but the PR was not up to date. I created a new one with new code but on my repo by mistake https://github.com/adoy/php-src/pull/1. I just pushed the new code and applied some of your comments to it since they were still valid. The new PR contains setters and getters instead of a "generic" one.